pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -117,16 +118,25 @@ func (r *ReferenceAdapter) EnsureProjectClaimUpdated() (gcpv1alpha1.ClaimStatus,
 
 	if r.projectClaim.Spec.GCPProjectID == "" {
```

Comment:
we could put this into its own function with similar signature than the function that adds availability zones, so we have a single level of abstraction in this function:
```
if  !ready:
  return

modified = ensureClaimProjectIdSet()
modified = modified || ensureClaimAvailabilityZonesSet()
if modified:
  updateClaim()
...
```